src/main/kotlin/com/team4099/robot2023/subsystems/drivetrain/drive/Drivetrain.kt

```diff
@@ -58,6 +60,17 @@ class Drivetrain(val gyroIO: GyroIO, swerveModuleIOs: DrivetrainIO) : SubsystemB
   val gyroInputs = GyroIO.GyroIOInputs()
   val swerveModules = swerveModuleIOs.getSwerveModules() // FL, FR, BL, BR
 
+  companion object {
+    var odometryLock: Lock = ReentrantLock()
+    fun setOdometryLock(Locked: Boolean) {
```

Comment:
not necessary just call lock and unlock

Reply:
is companion cause you need setOdometryLock to be static, otherwise you get error in cause its not static so you can't call the function without an object of the class